gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -58,12 +67,32 @@ public void apply(Project rootProject) {
         JdksExtension jdksExtension = JdksPlugin.extension(rootProject, jdkDistributions);
 
         rootProject.getPluginManager().withPlugin("com.palantir.baseline-java-versions", unused -> {
-            rootProject
-                    .getExtensions()
-                    .getByType(BaselineJavaVersionsExtension.class)
-                    .getSetupJdkToolchains()
-                    .set(false);
+            BaselineJavaVersionsExtension baselineJavaVersionsExtension =
+                    rootProject.getExtensions().getByType(BaselineJavaVersionsExtension.class);
+            baselineJavaVersionsExtension.getSetupJdkToolchains().set(false);
+
+            jdksExtension.jdkMajorVersionsToUse().set(rootProject.provider(() -> Stream.of(
+                            jdksExtension.getDaemonTarget(),
+                            baselineJavaVersionsExtension.libraryTarget(),
+                            baselineJavaVersionsExtension.runtime().map(ChosenJavaVersion::javaLanguageVersion),
+                            baselineJavaVersionsExtension
+                                    .distributionTarget()
+                                    .map(ChosenJavaVersion::javaLanguageVersion))
+                    .map(Provider::get)
+                    .collect(Collectors.toSet())));
         });
+
+        rootProject.subprojects(
+                proj -> proj.getPluginManager().withPlugin("com.palantir.baseline-java-version", unused -> {
+                    BaselineJavaVersionExtension projectVersions =
+                            proj.getExtensions().getByType(BaselineJavaVersionExtension.class);
+
+                    jdksExtension.jdkMajorVersionsToUse().addAll(rootProject.provider(() -> Stream.of(
+                                    projectVersions.target(), projectVersions.runtime())
```

Comment:
hmm, I don't believe this needs to be a provider no?

Reply:
oh, it's nested

hmm, can we just have a small utility method that's like:

```java
private static Provider<JavaLanguageVersion> asJavaLanguageVersion(Property<ChosenJavaVersion> version) {
  return version.map(ChosenJavaVersion::javaLanguageVersion)
}
```

Then usage is just:

```java
jdksExtension.jdjkMajorVersionsToUse().addAll(
  asJavaLanguageVersion(projectVersions.target()),
  asJavaLanguageVersion(projectVersions.runtime())
)
```

Feels like it's a bit easier to see what's going on